dll/win32/browseui/globalfoldersettings.cpp

```diff
@@ -20,6 +20,35 @@
 
 #include "precomp.h"
 
+#define ROSSIG ( ('R' << 0) | ('O' << 8) | ('S' << 16) | (('g' ^ 'f' ^ 's') << 24) )
+#define DEFBROWSERSTREAM L"Settings"
+
+template<class S, class D> static void CopyTo(const S &Src, D &Dst)
+{
+    Dst.FolderSettings = Src.FolderSettings;
+}
+
+static void EnsureValid(FOLDERSETTINGS &fs)
+{
+    if ((int)fs.ViewMode < FVM_AUTO || (int)fs.ViewMode > FVM_LAST)
```

Comment:
Cast is necessary?

```suggestion
    if (fs.ViewMode < FVM_AUTO || fs.ViewMode > FVM_LAST)
```

Reply:
Yes, FVM_AUTO is negative and FOLDERSETTINGS::ViewMode is UINT for some reason.